src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AppointmentService.kt

```diff
@@ -0,0 +1,67 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Appointment
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Attended
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.AppointmentRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.AuthUserRepository
+import java.time.OffsetDateTime
+import java.util.UUID
+import javax.transaction.Transactional
+
+@Service
+@Transactional
+class AppointmentService(
+  val appointmentRepository: AppointmentRepository,
+  val authUserRepository: AuthUserRepository,
+) {
+  fun createOrUpdateAppointment(
+    appointment: Appointment?,
+    durationInMinutes: Int,
+    appointmentTime: OffsetDateTime,
+    createdByUser: AuthUser
+  ): Appointment {
+    val noAppointment = appointment == null
+    if (noAppointment) {
+      return createAppointment(durationInMinutes, appointmentTime, createdByUser)
+    }
+
+    val appointmentWithoutAttendanceRecorded = appointment!!.attended == null
+    if (appointmentWithoutAttendanceRecorded) {
+      return updateAppointment(appointment, durationInMinutes, appointmentTime)
+    }
+
+    val appointmentWithNonAttendance = appointment.attended == Attended.NO
+    if (appointmentWithNonAttendance) {
+      return createAppointment(durationInMinutes, appointmentTime, createdByUser)
+    }
+    throw IllegalStateException("Is it not possible to update an appointment that has already been attended")
+  }
+
+  private fun createAppointment(
```

Comment:
small thing - i would expect this to be the place where the delius bookings are updated. is that coming later?

Reply:
Rajiv has created a branch with the booking work